api/types/types.proto

```diff
@@ -2609,6 +2609,41 @@ message GithubConnectorSpecV3 {
     string Display = 5 [ (gogoproto.jsontag) = "display" ];
 }
 
+// SSOInfoType represents the SSO diagnostic info type.
+enum SSOInfoType {
+    // Unknown info type.
+    UNKNOWN = 0;
+
+    // Error info type, used for reporting errors.
+    ERROR = 1;
+    // Success info type, used for marking successful flows.
+    SUCCESS = 2;
+    // Create user params, contains info about attempted user creation parameters.
+    CREATE_USER_PARAMS = 3;
+
+    // [SAML] Attributes to roles mapping.
+    SAML_ATTRIBUTES_TO_ROLES = 101;
+    // [SAML] Attributes to roles warnings.
+    SAML_ATTRIBUTES_TO_ROLES_WARNINGS = 102;
+    // [SAML] Assertion info.
+    SAML_ASSERTION_INFO = 103;
+    // [SAML] Attribute statements.
+    SAML_ATTRIBUTE_STATEMENTS = 104;
```

Comment:
What are these "types"? It does look a bit like we're trying to replace properly typed structs with this types enum + interface.